src/pages/Lend/LendByCategoryPage.vue

```diff
@@ -1,5 +1,6 @@
 <template>
 	<www-page
+		v-show="showContent"
```

Comment:
Can we instead do this redirect and exp check in the `preFetch`? See `LendFilterPage.vue` for an example of this:

```
preFetch(config, client, { cookieStore, route }) {
			// prefetch page data + experiment settings
			return client.query({
				query: lendFilterPageQuery,
				variables: {
					basketId: cookieStore.get('kvbskt')
				},
			}).then(() => {
				// check route for eligibility
				if (isFLSSEligible(route)) {
					return getExperimentSettingAsync(client, lendFilterRedirectEXP)
						.then(() => {
							// Assign the exp
							return client.query({
								query: experimentAssignmentQuery,
								variables: { id: lendFilterRedirectEXP }
							});
						}).then(({ data }) => {
							// Rediect if we are in the experiment group
							if (data?.experiment?.version === 'b') {
								return Promise.reject({
									path: '/lend/filter-alpha',
									query: route.query,
								});
							}
						});
				}
				// Otherwise just resolve
				Promise.resolve();
			});
		}
```

Reply:
Roger and I have discussed trying to do a prefetch and while I do think it's possible, it's also ok as a first pass to do all of the redirecting "and tracking" together in the single init method below...It would be good to do a little more work and get this promise chain in place as shown above though.

That said @roger-in-kiva I do not think we should hide ALL of the content. We can show a loader over all of the in page content while your exp is bootstrapped but not hide everything.

Reply:
Extending the promise chain and redirecting on the server/page load is my preference...

I think we can use `my.userAccount.id` field a conditional on the experiment assignment and then if the assignment is present decide about the redirect...

```
	preFetch(config, client) {
			return client.query({
				query: lendByCategoryQuery
			}).then(({ data })=> {
				// Get the array of channel objects from settings
				return Promise.all([
					// experiment: GROW-330 Machine Learning Category row
					client.query({ query: experimentQuery, variables: { id: 'EXP-ML-Service-Bandit-LendByCategory' } }),
					// experiment: CORE-698 MFI Recommendations
					client.query({ query: experimentQuery, variables: { id: 'mfi_recommendations' } }),
					// experiment: VUE- Category Service driven FLSS channels
					client.query({ query: experimentQuery, variables: { id: 'flss_category_service' } }),
					// experiment: CORE-854 Loan Finding Page
					data?.my?.userAccount?.id ? client.query({ query: experimentQuery, variables: { id: 'loan_finding_page' } }) : Promise.resolve(),
				]);
			}).then(expAssignments => {
               // check for loan_finding_page exp Assignment
               const targetExp = expAssignments?.this?.that?.exp?.version...
               if (targetExp === 'b') {
				  return Promise.reject({
					path: '/lending-home',
				  });
			   }
               Promise.resolve();
            })
				.then(() => {
					return client.query({
						query: mlOrderedLoanChannels
```

Reply:
In this case, will the puppet entry be required?

Reply:
This sounds good, but what about tracking events. Should they passed to the new page when created?

```
this.$kvTrackEvent(
       'Lending',
       'EXP-CORE-854-Dec2022',
	loanFindingEXP.version
);
```

Reply:
There are probably a couple options, and you'll want to test which is working like expected. If traffic only hits the new page from this redirect, then tracking on the new page should work fine. But if we only want to track if the user gets redirect (as opposed to opening the new page directly), then you'll want to do something similar to what's done in the `mounted` of `LendFilterPage.vue`.

Reply:
Tracking events on exps are generic in the sens that that they always fire if the exp is assigned and the version matches the context.

As Casey mentioned, you'd have a block in mounted to get the exp assignment and then fire the tracking for the assignment.

For LBC, where the exp is originally assigned, you'd use `getExperimentSettingCached` + `trackExperimentVersion` but on the new test page you'd need to use `getExperimentSettingAsync` + `trackExperimentVersion` to ensure you've fetched the setting and assignment.

Reply:
updated! WDYT @dyersituations @mcstover ?

Reply:
@roger-in-kiva I think this is looking great. Please remove this v-show line though so we don't hide/flash the entire page of content. You shouldn't need the new `showContent` mechanism at all now that you've wired up the prefetch!

Reply:
Looks good, and agreed regarding removing `showContent`

Reply:
removed, thanks guys!